pkg/types/types_test.go

```diff
@@ -0,0 +1,473 @@
+			}`,
+			leaf:          &RekorLeaf{},
+			expectSuccess: false,
+		},
+	}
+
+	for _, tc := range tests {
+		if tc.leaf == nil {
+			leaf, err := ParseRekorLeaf(strings.NewReader(tc.json))
+			if err != nil {
+				t.Errorf("unexpected error in '%v': %v", tc.caseDesc, err)
+			}
+			tc.leaf = &leaf
+		}
+
+		if _, err := ParseRekorEntry(strings.NewReader(tc.json), *(tc.leaf)); (err == nil) != tc.expectSuccess {
+			t.Errorf("Error in test case '%v': %v", tc.caseDesc, err)
+		}
+	}
+
+	if _, err := ParseRekorEntry(&BadReader{}, RekorLeaf{}); err == nil {
+		t.Errorf("No error thrown if io.Reader failed")
+	}
+
+	// valid leaf, invalid JSON
+	leaf, err := ParseRekorLeaf(strings.NewReader(tests[0].json))
+	if err != nil {
+		t.Errorf("unexpected error in '%v': %v", tests[0].caseDesc, err)
+	}
+
+	if _, err := ParseRekorEntry(strings.NewReader("not json"), leaf); err == nil {
+		t.Errorf("No error thrown for invalid JSON but valid leaf")
+	}
+}
+
+func TestRekorLoad(t *testing.T) {
+	type Test struct {
+		caseDesc      string
+		entry         *RekorEntry
+		expectSuccess bool
+	}
+
+	invalidFile := []byte("doesn't match")
+
+	validSHA := "0ddda6cfaa85ae5771cbbffc7d1fdbf4cfa6761c7256a642f91533363bf1d06f"
+	validSig, _ := base64.StdEncoding.DecodeString("LS0tLS1CRUdJTiBQR1AgU0lHTkFUVVJFLS0tLS0KQ29tbWVudDogVGhpcyBzaWduYXR1cmUgaXMgZm9yIHRoZSAudGFyIHZlcnNpb24gb2YgdGhlIGFyY2hpdmUKQ29tbWVudDogZ2l0IGFyY2hpdmUgLS1mb3JtYXQgdGFyIC0tcHJlZml4PWxpbnV4LTUuOC45LyB2NS44LjkKQ29tbWVudDogZ2l0IHZlcnNpb24gMi4yOC4wCgppUUl6QkFBQkNBQWRGaUVFWkg4b1pVaVU0NzFGY1ptK09OdTl5R0NTYVQ0RkFsOWN2YWdBQ2drUU9OdTl5R0NTCmFUNUMvZy8vWnZweTVMZHZHSUVYdkhIZ2xXck1CZDRPUy9ZYzZzbk84aDBZSWpGcnVIR1RQU3ZlbTZLRzByaEYKdy9McGV2aFhwc0IzUjh1N01LN2VNVG5rYVRkanpGbWhreldHZVptcW1PaWRMaGMrWm5WV012dHNnUjJTRXVPNApWd2Vzb2ZvSlkybmswdXhqNW1CcE1ZNkFsRG9oanlteWZzY3diSVNqNkZobTFpcXlNOGpWV3JxVU40WDZCV2dkCnBVS2ovUWRjcmdkS0FHRkpCMGJVeFEyR2I5aHczWmhrQndQRWNhZ2hKNU53Rm44ZzRJTm96UTY2RTBTbEdTK3cKSVNOZEh1d2x6VVg3WitsaUYvVUpsdkllV1ZSZTl3SUNkZnhRSTJqL2RJK1hqblc0VkNoWkxFU3dJeWxadDhqVwpaeDk2d1dPOHlFUmNrd1NablpWSjdrS3dBdmxRVitlNEQ3NkZZQUUzWURySThsTnFuL1hLTXhrZEpqS2hIVGNNCm8waXc3TVFuM2ZFdWxRNFVEd0c3UGEzTHNKeExxMkRVdkI4bG12bng3OG11VDZKTnVhOVVuYzNFTlptbER6YzcKb0pVcUlLeThkWTZZWFFCTWNNcUpoNUR1UkRWRXdKNm55WVlWMTRzWm1rRzgvV29mUkdReHVvY1JjT1RCQk12SgppZkRRb2IyWVdpRUd0UEpIdHh2VWI5RTZ4YnJoR2lscjJncmRDRHFMUUF3QkZTeWlCSy9Kc1dHTE5aMThnUEF6CmJlMGlYMk5mYUxlQlFKZUE2QjMxZk5TcStndHcvMzRBemVOTnNkT1BuZXlYcXN0OUs1WlpCWnFwMUMvZTljWFcKMS9YTlhWeXUwelp1Mkc2VHFlVFpyT3R4NVM4bEVvUjFpM09yQXN0Z0hGT2FibGRFc0VBPQo9WU14VgotLS0tLUVORCBQR1AgU0lHTkFUVVJFLS0tLS0K")
```

Comment:
ick, having all of these strings makes me worry about maintenance/debugging here. How many unique ones are there in total? Could we use files in testdata/ like the other test?

Reply:
yeah, I can clean this up.